invokeai/backend/ip_adapter/ip_adapter.py

```diff
@@ -192,24 +198,46 @@ def _init_image_proj_model(self, state_dict):
         ).to(self.device, dtype=self.dtype)
 
 
+def load_ip_adapter_tensors(ip_adapter_ckpt_path: str, device: str) -> IPAdapterStateDict:
+    state_dict: IPAdapterStateDict = {"ip_adapter": {}, "image_proj": {}}
+
+    if ip_adapter_ckpt_path.endswith("safetensors"):
+        model = safe_open(ip_adapter_ckpt_path, device=device, framework="pt")
+        for key in model.keys():
```

Comment:
I suspect that there are also IP-Adapter models saved in safetensors format with the same key structure as `ip_adapter.bin`. Fine with me if you want to wait until we encounter one in the wild though 🙂

Reply:
What is causing the change in the dict structure? The saving to safetensors? In which case, any bin model converted to safetensors using the native lib should always have the same structure -- unless some madman decides to mess with it. One battle at a time.

Reply:
I'm pretty sure the change in dict structure and saving to safetensors are totally unrelated. But yes, we can wait until this comes up